servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusDatabaseAdapterConfig.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.config;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithConverter;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
+import org.projectnessie.versioned.persist.nontx.NonTransactionalDatabaseAdapterConfig;
+
+@ConfigMapping(prefix = "nessie.adapter")
+@RegisterForReflection(targets = TrimmedStringConverter.class)
+public interface QuarkusDatabaseAdapterConfig extends NonTransactionalDatabaseAdapterConfig {
```

Comment:
How do we get the config for `TxDatabaseAdapterConfig` here?
I also would not mind to have a producer-bean for `NonTransactionalDatabaseAdapterConfig` and `TxDatabaseAdapterConfig` that takes Quarks-specific config instances via `@Inject`, if that's helpful.

Reply:
I was thinking we'd extend both Tx- and NonTransactional- config interfaces in one quarkus-based interfaces annotated with `@ConfigMapping`. Since all config variables are optionals/have defaults, it should not hurt to have all of them combined in one quarkus config group (i.e. `nessie.adapter`).

Another approach is to break down config groups into unrelated classes, I guess... something like
`@ConfigMapping(prefix = "nessie.adapter") QuarkusDatabaseAdapterConfig extends DatabaseAdapterConfig (extends Object)`:
* getKeyPrefix()
* etc....

`@ConfigMapping(prefix = "nessie.adapter.nontx") QuarkusNonTxDatabaseAdapterConfig extends NonTransactionalDatabaseAdapterConfig (extends Object)`:
* getParentsPerGlobalCommit()


`@ConfigMapping(prefix = "nessie.adapter.tx") QuarkusNonTxDatabaseAdapterConfig extends TxDatabaseAdapterConfig (extends Object)`:
* getBatchSize()

... but I personally like the approach where one quarkus class extends all adapter config classes. WDYT?

Reply:
I tend to agree (with "one thing extends all"). Maybe just have the `prefix`es so that we could split up later, if we need to?

Reply:
good point re: prefixes :+1:

Reply:
updated in e094dc26344a0f68b3d7c3adba03d0d7f118280d